src/ts/order.ts

```diff
@@ -160,13 +198,41 @@ export function hashify(h: HashLike): string {
 }
 
 /**
- * Normalized representation of an [`Order`] for EIP-712 operations.
+ * Normalizes the balance configuration for a buy token. Specifically, this
+ * function ensures that {@link OrderBalance.EXTERNAL} gets normalized to
+ * {@link OrderBalance.ERC20}.
+ *
+ * @param balance The balance configuration.
+ * @returns The normalized balance configuration.
  */
-export type NormalizedOrder = Omit<Order, "validTo" | "appData" | "kind"> & {
+export function normalizeBuyTokenBalance(
+  balance: OrderBalance | undefined,
+): OrderBalance.ERC20 | OrderBalance.INTERNAL {
+  switch (balance) {
+    case undefined:
+    case OrderBalance.ERC20:
+    case OrderBalance.EXTERNAL:
+      return OrderBalance.ERC20;
```

Comment:
I understand that the buy token balance can only be `ERC20` or `INTERNAL`. Would it be reasonable to throw an error in the case `OrderBalance.EXTERNAL`?
In case, I believe there would be no need to normalize anymore, because we could rely on `encodeFlag` throwing if a non-existing flag is detected.

Reply:
It was done like this since they technically mean the same thing. I had originally `type OrderBuyBalance = OrderBalance.ERC20 | OrderBalance.INTERNAL` but that caused a lot of code and type complexity in the tests and felt like it wasn't worth it.

Reply:
Created #615